app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlay.kt

```diff
@@ -0,0 +1,107 @@
+    override val hidesQuestTypes = setOf(
+        AddPathSurface::class.simpleName!!,
+        AddFootwayPartSurface::class.simpleName!!,
+        AddCyclewayPartSurface::class.simpleName!!,
+        AddSidewalkSurface::class.simpleName!!,
+    )
+
+    override fun getStyledElements(mapData: MapDataWithGeometry): Sequence<Pair<Element, Style>> =
+        mapData.filter( """
+            ways, relations with
+                highway ~ ${(ALL_PATHS).joinToString("|")}
+                or area != yes and (sidewalk ~ left|right|both or sidewalk:both = yes or sidewalk:left = yes or sidewalk:right = yes)
+            """).map { it to getStyle(it) }
+
+    /* cycleways with any sidewalk tagged on the way are common tagging for road-like cycleways in
+       Netherlands, when tapping on these, we want to be able to specify the sidewalk surface.
+       The cycleway (=road) surface can be tagged in the RoadSurfaceOverlay
+     */
+
+    override fun createForm(element: Element?): AbstractOverlayForm? {
+        val e = element ?: return null
+
+        val sidewalk = createSidewalkSides(e.tags)
+        val hasAnySidewalk = sidewalk?.left == Sidewalk.YES || sidewalk?.right == Sidewalk.YES
+        val isArea = element.tags["area"] == "yes"
+
+        return when {
+            !isArea && hasAnySidewalk -> SidewalkSurfaceOverlayForm()
+            else ->                      PathSurfaceOverlayForm()
+        }
+    }
+}
+
+private fun getStyle(element: Element): Style {
+    val sidewalk = createSidewalkSides(element.tags)
+    val hasAnySidewalk = sidewalk?.left == Sidewalk.YES || sidewalk?.right == Sidewalk.YES
+    val isArea = element.tags["area"] == "yes"
+
+    if (!isArea && hasAnySidewalk) {
+        val sidewalkSurface = createSidewalkSurface(element.tags)
+
+        val leftColor = getSidewalkColor(sidewalk?.left, sidewalkSurface?.left, element)
+        val rightColor = getSidewalkColor(sidewalk?.right, sidewalkSurface?.right, element)
+
+        return PolylineStyle(
+            stroke = null,
+            strokeLeft = StrokeStyle(leftColor),
+            strokeRight = StrokeStyle(rightColor)
+        )
+    } else {
+        val isSegregated = element.tags["segregated"] == "yes"
+        val color = if (isSegregated) {
+            val footwayColor = createSurfaceAndNote(element.tags, "footway").getColor(element)
+            val cyclewayColor = createSurfaceAndNote(element.tags, "cycleway").getColor(element)
+            // take worst case for showing
+            listOf(footwayColor, cyclewayColor).minBy { color ->
+                when (color) {
+                    Color.DATA_REQUESTED -> 0
+                    Color.INVISIBLE -> 1
+                    Color.BLACK -> 2
+                    else -> 3
+                }
+            }
+        } else {
+            createSurfaceAndNote(element.tags).getColor(element)
+        }
+        return if (isArea) PolygonStyle(color) else PolylineStyle(StrokeStyle(color))
+    }
```

Comment:
`else` is not needed after `return`

```suggestion
    }

    val isSegregated = element.tags["segregated"] == "yes"
    val color = if (isSegregated) {
        val footwayColor = createSurfaceAndNote(element.tags, "footway").getColor(element)
        val cyclewayColor = createSurfaceAndNote(element.tags, "cycleway").getColor(element)
        // take worst case for showing
        listOf(footwayColor, cyclewayColor).minBy { color ->
            when (color) {
                Color.DATA_REQUESTED -> 0
                Color.INVISIBLE -> 1
                Color.BLACK -> 2
                else -> 3
            }
        }
    } else {
        createSurfaceAndNote(element.tags).getColor(element)
    }
    return if (isArea) PolygonStyle(color) else PolylineStyle(StrokeStyle(color))
```

Reply:
true but i find it better this way